templates/commands/render_action_append.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+
+	"github.com/abcxyz/abc/templates/model"
+)
+
+func actionAppend(ctx context.Context, ap *model.Append, sp *stepParams) error {
```

Comment:
Would like feedback on use of parse&execute and walk&modify. I based off of what I saw in the string replace action, but don't fully understand why I'm parsing a go template when its not really a template.

Reply:
We use go-templates here so the spec.yaml can use different filenames depending on the input. Like:

```
action: 'append'
params:
  path: 'dbconfigs/{{.dbname}}.cfg
  with: ...
```

We try to support go-template expansion in as many places as possible in the spec file.